src/wallet/components/WalletDropdownFundLink.tsx

```diff
@@ -1,34 +1,79 @@
+import { useCallback, useMemo } from 'react';
 import { useGetFundingUrl } from '../../fund/hooks/useGetFundingUrl';
+import { getPopupSize } from '../../fund/utils/getPopupSize';
+import { openPopup } from '../../internal/utils/openPopup';
+import { cn, pressable, text as themeText } from '../../styles/theme';
+import { useIcon } from '../hooks/useIcon';
 import type { WalletDropdownFundLinkReact } from '../types';
-import { WalletDropdownFundLinkButton } from './WalletDropdownFundLinkButton';
 
 export function WalletDropdownFundLink({
+  className,
   fundingUrl,
-  ...props
+  icon = 'fundWallet',
+  openIn = 'popup',
+  popupSize = 'md',
+  rel,
+  target,
+  text = 'Fund wallet',
 }: WalletDropdownFundLinkReact) {
-  const defaultFundingUrl = useGetFundingUrl();
+  // If we can't get a funding URL, this component will be a no-op and render a disabled link
+  const fundingUrlToRender = fundingUrl ?? useGetFundingUrl();
+  const iconSvg = useIcon({ icon });
 
-  // If the fundingUrl prop is undefined, fallback to the default funding URL
-  const fundingUrlToRender = fundingUrl ?? defaultFundingUrl?.url;
-  const popupHeightOverride = fundingUrl
-    ? undefined
-    : defaultFundingUrl?.popupHeight;
-  const popupWidthOverride = fundingUrl
-    ? undefined
-    : defaultFundingUrl?.popupWidth;
+  const handleClick = useCallback(
+    (e: React.MouseEvent) => {
+      e.preventDefault();
+      if (fundingUrlToRender) {
+        const { height, width } = getPopupSize(popupSize, fundingUrlToRender);
+        openPopup({
+          url: fundingUrlToRender,
+          height,
+          width,
+          target,
+        });
+      }
+    },
+    [fundingUrlToRender, popupSize, target],
+  );
 
-  if (fundingUrlToRender) {
+  const overrideClassName = cn(
+    pressable.default,
+    // Disable hover effects if there is no funding URL
+    !fundingUrlToRender && 'pointer-events-none',
```

Comment:
with this change, https://github.com/coinbase/onchainkit/pull/1322/files#diff-639b39dc7c8fa5ec11491e030f9add3c94620cae4c45a8e0a9412103659e41dfR31, this shouldn't be necessary right?

Reply:
@alessey na it is necessary. I don't use `pressable.disabled` on the actual link because that would make the background color change like this:

![Screenshot 2024-10-01 at 6 36 25 AM](https://github.com/user-attachments/assets/d87d34ce-8bdf-4ee0-82b5-e3431b24546e)

So I disable the pointer events on the link to prevent the hover effect from `pressable.default` which is on the link, and then apply `pressable.disabled` to the span inside the link so that the text is dimmed to look disabled.